pkg/controllers/status/controller.go

```diff
@@ -233,6 +233,17 @@ func NewStatusController(
 		return nil, fmt.Errorf("failed to add cluster event handler: %w", err)
 	}
 
+	if err := s.ftcManager.AddFTCUpdateHandler(func(lastObserved, latest *fedcorev1a1.FederatedTypeConfig) {
+		if latest == nil || !latest.IsStatusCollectionEnabled() {
```

Comment:
I think this overlaps with the `Predicate` passed to `AddEventHandlerGenerators` above. `AddEventHandlerGenerator` will also trigger a reenqueue of all objects as the newly registered eventhandler will receive an add event for all existing objects in the store.

Reply:
We probably only need to keep the `!reflect.DeepEqual(lastObserved.Spec.StatusCollection.Fields, latest.Spec.StatusCollection.Fields)` check here. Alternatively, we can move this check to the `Predicate`s above.